### PR TITLE
[sensor_ctrl, ast] Preparation for sensor_ctrl d2

### PR DESCRIPTION
**Pull request by tjaychen**
- Add sensor_ctrl doc and prj.hjson
- Update ast title header
- Minor fix for sensor_ctrl alert clearing

Signed-off-by: Timothy Chen <timothytim@google.com>

### CODE REVIEW

File: hw/top_earlgrey/ip/ast/doc/_index.md [hunk 1]
```diff
@@ -1,3 +1,7 @@
+---
+title: "Analog Sensor Top Technical Specification"
```
Comment (tjaychen) [inline hw/top_earlgrey/ip/ast/doc/_index.md:2]: @zi-v 
this is a minor change that works better with opentitan's document generation.
Let me know if this is okay with you.

Also I guess we should do an doc update for the recently added ports?

Comment (zi-v) [reply hw/top_earlgrey/ip/ast/doc/_index.md:2]: OK

File: hw/top_earlgrey/ip/sensor_ctrl/doc/_index.md [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+- Alert hand-shake with `analog sensor top`
+- Alert forwarding to `alert handler`
+- Status readback for `analog sensor top`
+- Pad debug hook up for `analog sensor top`
+
+# Theory of Operations
+
+## Block Diagram
+
+The diagram below shows how `sensor control` helps `analog sensor top` integration into the overall design.
```
Comment (cindychip) [inline hw/top_earlgrey/ip/sensor_ctrl/doc/_index.md:24]: I assume you will add it later right?

**Review by eunchan — APPROVED** [review]

**Comment by tjaychen** [comment]
o crap, i didn't realize i forgot to do that.
Thanks for pointing it out.  I'll add it later today.

File: hw/top_earlgrey/ip/sensor_ctrl/data/sensor_ctrl.prj.hjson [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+    name:               "sensor_ctrl",
+    design_spec:        "../doc",
+    dv_doc:             "../doc/dv",
+    hw_checklist:       "../doc/checklist",
+    revisions: [
+      {
+        version:            "0.1",
+        life_stage:         "L1",
+        //design_stage:       "D3",
+        //verification_stage: "V3",
```
Comment (sriyerg) [inline hw/top_earlgrey/ip/sensor_ctrl/data/sensor_ctrl.prj.hjson:15]: Should these be D1 / V0?

File: hw/top_earlgrey/ip/sensor_ctrl/doc/_index.md [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+
+## Alert Handshake
+
+The `analog sensor top` sends alert requests in independent, differential form to the `sensor control`.
+Each alert request is consisted of a pair of signals, one active high and one active low.
```
Comment (sriyerg) [inline hw/top_earlgrey/ip/sensor_ctrl/doc/_index.md:29]:
```suggestion
Each alert request consists of a pair of signals, one active high and one active low.
```

File: hw/top_earlgrey/ip/sensor_ctrl/doc/_index.md [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+However, the active polarity of each signal is independent, this reflects the imprecise timing sensor that drives the alert.
```
Comment (sriyerg) [inline hw/top_earlgrey/ip/sensor_ctrl/doc/_index.md:30]: nit: two unrelated sentences seem to be joined with a comma

File: hw/top_earlgrey/ip/sensor_ctrl/doc/_index.md [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+This means that the `sensor control` recognizes an active alert as long as one of the lines is active, and not the pair of signals being in a particular state.
```
Comment (sriyerg) [inline hw/top_earlgrey/ip/sensor_ctrl/doc/_index.md:31]: Perhaps say "each signal in the differential pair is a separate dedicated alert signal" to make this clear.

File: hw/top_earlgrey/ip/sensor_ctrl/doc/_index.md [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+Once an alert request is detected as active, the `sensor control` then formulates a proper alert event through the `prim_alert_sender` and sends a notification to the `alert handler`.
+
+The `sensor control` can optionally generate alert acknowledgements back to the `analog sensor top`.
+When an alert request is detected as active, the `sensor control` has 3 options after sending out the alert:
+
+- Do not acknowledge, thus continually triggering more and more alerts
+- Software acknowledge. This allows software to decide whether an alert should stop triggering.
+- Hardware acknowledge. The hardware immediately acknowledges the alert.
```
Comment (sriyerg) [inline hw/top_earlgrey/ip/sensor_ctrl/doc/_index.md:39]:
```suggestion
- Do not acknowledge: Thus, let more and more alerts to be continuously triggered.
- Software acknowledge: Let the software to decide whether an alert should stop triggering.
- Hardware acknowledge: Let the hardware immediately acknowledges the alert.
```

**Review by sriyerg — COMMENTED** [review]
Left a few mostly nitpicky comments. LGTM overall.

File: hw/top_earlgrey/ip/sensor_ctrl/data/sensor_ctrl.prj.hjson (hunk 3 above)
Comment (tjaychen) [reply hw/top_earlgrey/ip/sensor_ctrl/data/sensor_ctrl.prj.hjson]: you're right! thanks.

File: hw/top_earlgrey/ip/sensor_ctrl/doc/_index.md (hunk 4 above)
Comment (tjaychen) [reply hw/top_earlgrey/ip/sensor_ctrl/doc/_index.md]: updated

File: hw/top_earlgrey/ip/sensor_ctrl/doc/_index.md (hunk 5 above)
Comment (tjaychen) [reply hw/top_earlgrey/ip/sensor_ctrl/doc/_index.md]: updated

File: hw/top_earlgrey/ip/sensor_ctrl/doc/_index.md (hunk 6 above)
Comment (tjaychen) [reply hw/top_earlgrey/ip/sensor_ctrl/doc/_index.md:31]: updated

File: hw/top_earlgrey/ip/sensor_ctrl/doc/_index.md (hunk 7 above)
Comment (tjaychen) [reply hw/top_earlgrey/ip/sensor_ctrl/doc/_index.md]: updated

File: hw/top_earlgrey/ip/sensor_ctrl/data/sensor_ctrl.prj.hjson [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+    design_spec:        "../doc",
+    dv_doc:             "../doc/dv",
+    hw_checklist:       "../doc/checklist",
+    revisions: [
+      {
+        version:            "0.1",
+        life_stage:         "L1",
+        design_stage:       "D1",
+        verification_stage: "V0",
+        //notes:              ""
```
Comment (sriyerg) [inline hw/top_earlgrey/ip/sensor_ctrl/data/sensor_ctrl.prj.hjson:16]:
```suggestion
        notes:              "Verified directly at the chip level"
```

**Comment by sriyerg** [comment]
Hi Tim, you probably overlooked, but there are a few outstanding comments.

**Comment by tjaychen** [comment]
> Hi Tim, you probably overlooked, but there are a few outstanding comments.

o did i miss one? i thought i addressed them.

**Pull request merged**